Userland/Applications/Calendar/CalendarWidget.cpp

```diff
@@ -0,0 +1,308 @@
+/*
+ * Copyright (c) 2023, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "CalendarWidget.h"
+#include "AddEventDialog.h"
+#include <AK/JsonParser.h>
+#include <AK/LexicalPath.h>
+#include <Applications/Calendar/CalendarWindowGML.h>
+#include <LibConfig/Client.h>
+#include <LibCore/System.h>
+#include <LibGUI/Action.h>
+#include <LibGUI/ActionGroup.h>
+#include <LibGUI/Application.h>
+#include <LibGUI/BoxLayout.h>
+#include <LibGUI/Calendar.h>
+#include <LibGUI/Icon.h>
+#include <LibGUI/Menu.h>
+#include <LibGUI/Menubar.h>
+#include <LibGUI/Painter.h>
+#include <LibGUI/Process.h>
+#include <LibGUI/Toolbar.h>
+#include <LibGUI/Window.h>
+#include <LibGfx/Font/FontDatabase.h>
+#include <LibGfx/Palette.h>
+#include <LibMain/Main.h>
+
+namespace Calendar {
+
+ErrorOr<NonnullRefPtr<CalendarWidget>> CalendarWidget::create(GUI::Window* parent_window)
+{
+    auto widget = TRY(AK::adopt_nonnull_ref_or_enomem(new (nothrow) CalendarWidget));
+    TRY(widget->load_from_gml(calendar_window_gml));
+
+    widget->m_calendar_frame = widget->find_descendant_of_type_named<GUI::Calendar>("calendar");
+    widget->m_calendar = Calendar::create();
+    widget->create_on_events_change();
+
+    auto toolbar = widget->find_descendant_of_type_named<GUI::Toolbar>("toolbar");
+    auto calendar = widget->m_calendar_frame;
+
+    auto prev_date_action = TRY(widget->create_prev_date_action());
+    auto next_date_action = TRY(widget->create_next_date_action());
+
+    auto add_event_action = TRY(widget->create_add_event_action());
+
+    auto jump_to_action = TRY(widget->create_jump_to_action());
+
+    auto view_month_action = TRY(widget->create_view_month_action());
+    view_month_action->set_checked(true);
+
+    auto view_year_action = TRY(widget->create_view_year_action());
+    auto view_type_action_group = make<GUI::ActionGroup>();
+
+    view_type_action_group->set_exclusive(true);
+    view_type_action_group->add_action(*view_month_action);
+    view_type_action_group->add_action(*view_year_action);
+    auto default_view = Config::read_string("Calendar"sv, "View"sv, "DefaultView"sv, "Month"sv);
+    if (default_view == "Year")
+        view_year_action->set_checked(true);
+
+    auto open_settings_action = TRY(widget->create_open_settings_action());
+
+    (void)TRY(toolbar->try_add_action(prev_date_action));
+    (void)TRY(toolbar->try_add_action(next_date_action));
+    TRY(toolbar->try_add_separator());
+    (void)TRY(toolbar->try_add_action(jump_to_action));
+    (void)TRY(toolbar->try_add_action(add_event_action));
+    TRY(toolbar->try_add_separator());
+    (void)TRY(toolbar->try_add_action(view_month_action));
+    (void)TRY(toolbar->try_add_action(view_year_action));
+    (void)TRY(toolbar->try_add_action(open_settings_action));
```

Comment:
not sure if `(void)` is strictly needed here…? 🤔 (ignore if it is!)

Reply:
It's ignoring the `NonnullRefPtr` returned if `try_add_action` is successful